### PR TITLE
Add nationality model and edit database

### PR DESCRIPTION
**Pull request by hsiaotingluv**

### CODE REVIEW

File: src/test/java/seedu/address/model/person/NationalityTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public void isValidNationality() {
+        // null nationality
+        assertThrows(NullPointerException.class, () -> Nationality.isValidNationality(null));
+
+        // invalid nationality
+        assertFalse(Nationality.isValidNationality("")); // empty string
+        assertFalse(Nationality.isValidNationality(" ")); // spaces only
+
+        // valid nationality
+        assertTrue(Nationality.isValidNationality("Blk 456, Den Road, #01-355"));
```
Comment (kevinmingtarja) [inline src/test/java/seedu/address/model/person/NationalityTest.java:32]: not that big of a deal but the nationality is still an address

File: src/test/java/seedu/address/model/person/NameContainsKeywordsPredicateTest.java [hunk 2]
```diff
@@ -70,6 +70,6 @@ public void test_nameDoesNotContainKeywords_returnsFalse() {
         // Keywords match phone, email and address, but does not match name
         predicate = new NameContainsKeywordsPredicate(Arrays.asList("12345", "alice@email.com", "Main", "Street"));
         assertFalse(predicate.test(new PersonBuilder().withName("Alice").withPhone("12345")
-                .withEmail("alice@email.com").withAddress("Main Street").build()));
+                .withEmail("alice@email.com").withNationality("Main Street").build()));
```
Comment (kevinmingtarja) [inline src/test/java/seedu/address/model/person/NameContainsKeywordsPredicateTest.java:73]: Same comment

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 3]
```diff
@@ -104,25 +104,25 @@ public void parsePhone_validValueWithWhitespace_returnsTrimmedPhone() throws Exc
...
     public void parseAddress_invalidValue_throwsParseException() {
-        assertThrows(ParseException.class, () -> ParserUtil.parseAddress(INVALID_ADDRESS));
+        assertThrows(ParseException.class, () -> ParserUtil.parseNationality(INVALID_ADDRESS));
     }
 
     @Test
     public void parseAddress_validValueWithoutWhitespace_returnsAddress() throws Exception {
-        Address expectedAddress = new Address(VALID_ADDRESS);
-        assertEquals(expectedAddress, ParserUtil.parseAddress(VALID_ADDRESS));
+        Nationality expectedNationality = new Nationality(VALID_ADDRESS);
```
Comment (kevinmingtarja) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:117]: Same here, the nationality still takes in `VALID_ADDRESS`

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 4]
```diff
@@ -104,25 +104,25 @@ public void parsePhone_validValueWithWhitespace_returnsTrimmedPhone() throws Exc
...
+        assertEquals(expectedNationality, ParserUtil.parseNationality(VALID_ADDRESS));
     }
 
     @Test
     public void parseAddress_validValueWithWhitespace_returnsTrimmedAddress() throws Exception {
         String addressWithWhitespace = WHITESPACE + VALID_ADDRESS + WHITESPACE;
-        Address expectedAddress = new Address(VALID_ADDRESS);
-        assertEquals(expectedAddress, ParserUtil.parseAddress(addressWithWhitespace));
+        Nationality expectedNationality = new Nationality(VALID_ADDRESS);
+        assertEquals(expectedNationality, ParserUtil.parseNationality(addressWithWhitespace));
```
Comment (kevinmingtarja) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:125]: Here as well

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 5]
```diff
@@ -20,22 +20,22 @@
...
-                new Address("Blk 436 Serangoon Gardens Street 26, #16-43"),
+                new Nationality("Blk 436 Serangoon Gardens Street 26, #16-43"),
                 getTagSet("family")),
             new Person(new Name("Irfan Ibrahim"), new Phone("92492021"), new Email("irfan@example.com"),
-                new Address("Blk 47 Tampines Street 20, #17-35"),
+                new Nationality("Blk 47 Tampines Street 20, #17-35"),
                 getTagSet("classmates")),
             new Person(new Name("Roy Balakrishnan"), new Phone("92624417"), new Email("royb@example.com"),
-                new Address("Blk 45 Aljunied Street 85, #11-31"),
+                new Nationality("Blk 45 Aljunied Street 85, #11-31"),
```
Comment (kevinmingtarja) [inline src/main/java/seedu/address/model/util/SampleDataUtil.java:38]: Same comment as well

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 6]
```diff
@@ -71,13 +71,13 @@ public static Phone parsePhone(String phone) throws ParseException {
      *
      * @throws ParseException if the given {@code address} is invalid.
```
Comment (bklimey) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:72]: `address` → `nationality`

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 7]
```diff
@@ -71,13 +71,13 @@ public static Phone parsePhone(String phone) throws ParseException {
...
      */
-    public static Address parseAddress(String address) throws ParseException {
+    public static Nationality parseNationality(String address) throws ParseException {
```
Comment (bklimey) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:74]: `address` → `nationality`

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 8]
```diff
@@ -153,8 +158,8 @@ public void parse_oneFieldSpecified_success() {
         assertParseSuccess(parser, userInput, expectedCommand);
 
         // address
```
Comment (bklimey) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java:160]: `address` → `nationality`

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 9]
```diff
@@ -103,26 +103,26 @@ public void parsePhone_validValueWithWhitespace_returnsTrimmedPhone() throws Exc
...
+        assertEquals(expectedNationality, ParserUtil.parseNationality(VALID_NATIONALITY));
     }
 
     @Test
-    public void parseAddress_validValueWithWhitespace_returnsTrimmedAddress() throws Exception {
-        String addressWithWhitespace = WHITESPACE + VALID_ADDRESS + WHITESPACE;
-        Address expectedAddress = new Address(VALID_ADDRESS);
-        assertEquals(expectedAddress, ParserUtil.parseAddress(addressWithWhitespace));
+    public void parseNationality_validValueWithWhitespace_returnsTrimmedNationality() throws Exception {
+        String addressWithWhitespace = WHITESPACE + VALID_NATIONALITY + WHITESPACE;
```
Comment (bklimey) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:123]: `addressWithWhitespace` → `nationalityWithWhitespace`

File: src/test/java/seedu/address/model/NationalityBookTest.java [hunk 10]
```diff
@@ -22,7 +22,7 @@
 import seedu.address.model.person.exceptions.DuplicatePersonException;
 import seedu.address.testutil.PersonBuilder;
 
-public class AddressBookTest {
+public class NationalityBookTest {
```
Comment (bklimey) [inline src/test/java/seedu/address/model/NationalityBookTest.java:25]: `NationalityBookTest` → `AddressBookTest` as `AddressBook` ≠ `Address`
Same for the java file name.

File: src/test/java/seedu/address/testutil/EditPersonDescriptorBuilder.java [hunk 11]
```diff
@@ -66,8 +66,8 @@ public EditPersonDescriptorBuilder withEmail(String email) {
     /**
      * Sets the {@code Address} of the {@code EditPersonDescriptor} that we are building.
```
Comment (bklimey) [inline src/test/java/seedu/address/testutil/EditPersonDescriptorBuilder.java:67]: `Address` → `Nationality`

File: src/test/java/seedu/address/testutil/EditPersonDescriptorBuilder.java [hunk 12]
```diff
@@ -66,8 +66,8 @@ public EditPersonDescriptorBuilder withEmail(String email) {
...
      */
-    public EditPersonDescriptorBuilder withAddress(String address) {
-        descriptor.setAddress(new Address(address));
+    public EditPersonDescriptorBuilder withNationality(String address) {
```
Comment (bklimey) [inline src/test/java/seedu/address/testutil/EditPersonDescriptorBuilder.java:69]: `address` → `nationality`

**Review by kevinmingtarja — APPROVED** [review]
LGTM, Thanks Hsiao Ting!

**Pull request merged**